showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITOtelMetrics.java

```diff
@@ -0,0 +1,554 @@
+            .toBuilder()
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .setTracerFactory(createOpenTelemetryTracerFactory("4326"))
+            .setTransportChannelProvider(
+                EchoSettings.defaultGrpcTransportProviderBuilder()
+                    .setChannelConfigurator(ManagedChannelBuilder::usePlaintext)
+                    .build())
+            .setEndpoint("localhost:7469")
+            .build();
+
+    EchoClient grpcClientWithRetrySetting = EchoClient.create(grpcEchoSettings);
+
+    BlockRequest blockRequest =
+        BlockRequest.newBuilder()
+            .setError(Status.newBuilder().setCode(Code.INVALID_ARGUMENT.ordinal()).build())
+            .build();
+
+    grpcClientWithRetrySetting.blockCallable().futureCall(blockRequest).isDone();
+
+    Thread.sleep(5000);
+
+    String filePath = "../directory10/testGrpc_attemptPermanentFailure_metrics.txt";
+    File file = new File(filePath);
+    Truth.assertThat(file.exists()).isTrue();
+    Truth.assertThat(file.length() > 0).isTrue();
+
+    grpcClientWithRetrySetting.close();
+    grpcClientWithRetrySetting.awaitTermination(
+        TestClientInitializer.AWAIT_TERMINATION_SECONDS, TimeUnit.SECONDS);
+  }
+
+  // Helper function for creating Opentelemetry object with a different port for exporter for every
+  // test
+  // this ensures that logs for each test are collected separately
+  private static ApiTracerFactory createOpenTelemetryTracerFactory(String port) {
+    // OTLP Metric Exporter setup
+    String endpoint = "http://localhost:" + port;
+    OtlpGrpcMetricExporter metricExporter =
+        OtlpGrpcMetricExporter.builder().setEndpoint(endpoint).build();
+
+    // Periodic Metric Reader configuration
+    PeriodicMetricReader metricReader =
+        PeriodicMetricReader.builder(metricExporter)
+            .setInterval(java.time.Duration.ofSeconds(3))
+            .build();
+
+    // OpenTelemetry SDK Configuration
+    Resource resource = Resource.builder().build();
+    SdkMeterProvider sdkMeterProvider =
+        SdkMeterProvider.builder().registerMetricReader(metricReader).setResource(resource).build();
+
+    OpenTelemetry openTelemetry =
+        OpenTelemetrySdk.builder().setMeterProvider(sdkMeterProvider).build();
+
+    // Meter Creation
+    Meter meter =
+        openTelemetry
+            .meterBuilder("gax")
+            .setInstrumentationVersion(GaxProperties.getGaxVersion())
+            .build();
+
+    // OpenTelemetry Metrics Recorder
+    OpentelemetryMetricsRecorder otelMetricsRecorder = new OpentelemetryMetricsRecorder(meter);
+
+    // Finally, create the Tracer Factory
+    return new MetricsTracerFactory(otelMetricsRecorder);
+  }
+
+  private void setupOtelCollector(String scriptPath, String directoryPath, String configPath)
+      throws Exception {
+    Process process =
+        Runtime.getRuntime().exec(scriptPath + " " + directoryPath + " " + configPath);
+    process.waitFor();
+  }
```

Comment:
I'm not sure that we should have the ITs create a new process or start the OTEL collector. I think  it would be better if this is run separately (i.e. manually for local testing).

The same way we do with the gapic-showcase server itself. We start up a local instance ourselves via (`gapic-showcase run`).

For the GH actions, I think you can create a step that run the startup scripts.

Reply:
We need a new otelCollector for every test, that's why.

Reply:
Having to create multiple otelCollectors is ok imo. My point above is that I don't think we should have a Java method that essentially executes a shell script in the test. I think it would be better to move all that otel initialization logic down to the ./startup_otel script and have that script create the necessary (10+ or however many) otel collectors

 > We need a new otelCollector for every test, that's why.

What about a single otelCollector for all the tests and the metrics in each test case have a unique identifier? Is something like that possible so that we won't need to create a new otel collector for every new test added?